packages/status-js/src/utils/encode-url-data.compare.test.ts

```diff
@@ -0,0 +1,643 @@
+import { utf8 } from '@scure/base'
+import { describe, expect, test } from 'vitest'
+
+import { encodeUrlData } from './encode-url-data.compare'
+
+import type {
+  CommunityChatPreview,
+  CommunityPreview,
+  UserPreview,
+} from '../protos/link-preview_pb'
+import type { PlainMessage } from '@bufbuild/protobuf'
+
+describe('community', () => {
+  test('A', () => {
+    const data = {
+      displayName: 'Lorem ipsum dolor sit egestas.',
+      description:
+        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus non dui vitae augue elementum laoreet ac pharetra odio. Morbi vestibulum.',
+      membersCount: 1_000_000,
+      color: '#4360DF',
+    }
+    const encodedData = encodeUrlData('community', data, {
+      serialization: 'csv',
+      compression: 'noop',
+      encoding: 'encodeURIComponent',
+    })
+    const characterLength = countCharacters(data)
+
+    expect(characterLength).toBe(184)
+    expect(encodedData).toBe(
+      'Lorem%20ipsum%20dolor%20sit%20egestas.%2CLorem%20ipsum%20dolor%20sit%20amet%2C%20consectetur%20adipiscing%20elit.%20Phasellus%20non%20dui%20vitae%20augue%20elementum%20laoreet%20ac%20pharetra%20odio.%20Morbi%20vestibulum.%2C1000000%2C%234360DF'
+    )
+    expect(encodedData).toHaveLength(243)
+  })
```

Comment:
I contemplated leaving these comparison test cases in the repo, but deciding against it.